src/components/learner-credit-management/data/hooks/useSubsidyAccessPolicy.ts

```diff
@@ -21,10 +23,32 @@ const getSubsidyAccessPolicy = async ({ queryKey }) => {
   return subsidyAccessPolicy;
 };
 
-const useSubsidyAccessPolicy = (subsidyAccessPolicyId) => useQuery({
-  queryKey: learnerCreditManagementQueryKeys.budget(subsidyAccessPolicyId),
-  queryFn: getSubsidyAccessPolicy,
-  enabled: !!subsidyAccessPolicyId,
-});
+const useSubsidyAccessPolicy = (subsidyAccessPolicyId) => {
+  const intl = useIntl();
+
+  return useQuery({
+    queryKey: learnerCreditManagementQueryKeys.budget(subsidyAccessPolicyId),
+    queryFn: getSubsidyAccessPolicy,
+    enabled: !!subsidyAccessPolicyId,
+    select: (data) => {
+      if (!data) {
+        return data;
+      }
+
+      // Calculate if the budget is retired or expired using getBudgetStatus for consistency
+      const { status } = getBudgetStatus({
+        intl,
+        startDateStr: data.subsidyActiveDatetime,
+        endDateStr: data.subsidyExpirationDatetime,
+        isBudgetRetired: data.retired,
+      });
+
+      return {
+        ...data,
+        isRetiredOrExpired: isBudgetRetiredOrExpired(status),
```

Comment:
Maybe long term outside of scope for this ticket: is it worth exposing a `isRetiredOrExpired` computed field on the serializer to migrate this business logic upstream?

Curious to the tradeoffs of adding a computed field from the API.